warehouse/upload.go

```diff
@@ -824,19 +828,76 @@ func (job *UploadJobT) updateTableSchema(tName string, tableSchemaDiff warehouse
 		return nil
 	}
 
-	if err := job.addColumnsToWarehouse(tName, tableSchemaDiff.ColumnMap); err != nil {
-		return err
+	if err = job.addColumnsToWarehouse(tName, tableSchemaDiff.ColumnMap); err != nil {
+		return fmt.Errorf("adding columns to warehouse: %w", err)
+	}
+
+	if err = job.alterColumnsToWarehouse(tName, tableSchemaDiff.AlteredColumnMap); err != nil {
+		return fmt.Errorf("altering columns to warehouse: %w", err)
 	}
 
-	for _, columnName := range tableSchemaDiff.StringColumnsToBeAlteredToText {
-		err = job.whManager.AlterColumn(tName, columnName, "text")
+	return nil
+}
+
+func (job *UploadJobT) alterColumnsToWarehouse(tName string, columnsMap map[string]string) error {
+	var responseToAlerta []model.AlterTableResponse
+	var errs []error
+
+	for columnName, columnType := range columnsMap {
+		res, err := job.whManager.AlterColumn(tName, columnName, columnType)
 		if err != nil {
-			pkgLogger.Errorf("Altering column %s in table: %s.%s failed. Error: %v", columnName, job.warehouse.Namespace, tName, err)
-			break
+			errs = append(errs, err)
+			continue
+		}
+
+		if res.IsDependent {
+			responseToAlerta = append(responseToAlerta, res)
+			continue
 		}
+
+		pkgLogger.Infof(`
+			[WH]: Altered column %s of type %s in table %s in namespace %s of destination %s:%s
+		`,
+			columnName,
+			columnType,
+			tName,
+			job.warehouse.Namespace,
+			job.warehouse.Type,
+			job.warehouse.Destination.ID,
+		)
 	}
 
-	return err
+	if len(responseToAlerta) > 0 {
+		queries := make([]string, len(responseToAlerta))
+		for i, res := range responseToAlerta {
+			queries[i] = res.Query
+		}
+
+		query := strings.Join(queries, "\n")
+		err := alerta.NewClient(
+			job.AlertaURL,
+			alerta.WithTeam(warehouseutils.WAREHOUSE),
+			alerta.WithTags(alerta.Tags{
+				fmt.Sprintf("destID=%s", job.upload.DestinationID),
+				fmt.Sprintf("destType=%s", job.upload.DestinationType),
+				fmt.Sprintf("query=%s", query),
+			}),
```

Comment:
Should the tags be part of the NewClient options?

Also, what if we use map[string]string instead of []string for tags, seems a better abstraction:  
```suggestion
			alerta.WithTags(alerta.Tags{
				"destID": job.upload.DestinationID,
			         "destType": job.upload.DestinationType,
			         "query": query,
			}),
```

Reply:
Actually for alerts Tags should be a list.
https://github.com/rudderlabs/alerta/blob/039c1fb04b48615efea54844ae0994404d16d4a8/alerta/models/alert.py#L121